### PR TITLE
Add a specific error for missing columns during materialization

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:
Follow up from #1615 based on the comment in #1614 . Also adds test and removed an empty file.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/feast/errors.py [hunk 1]
```diff
@@ -67,3 +69,11 @@ def __init__(self, offline_store_name: str, data_source_name: str):
         super().__init__(
             f"Offline Store '{offline_store_name}' does not support data source '{data_source_name}'"
         )
+
+
+class FeastJoinKeysDuringMaterialization(Exception):
+    def __init__(self, join_key_columns: Set[str], source_columns: Set[str]):
+        super().__init__(
+            f"The DataFrame must have at least {join_key_columns} columns present, "
```
Comment (woop) [inline sdk/python/feast/errors.py:77]: `The DataFrame` seems a bit unclear. Is it possible for us to be more specific?

Comment (achals) [reply sdk/python/feast/errors.py:77]: Yeah `The Dataframe being materialized`?

Comment (woop) [reply sdk/python/feast/errors.py]: I mean what is `The DataFrame`?

Comment (achals) [reply sdk/python/feast/errors.py]: Yeah added that

**Comment by achals** [comment]
@woop

**Comment by achals** [comment]
@woop for some reason this didn't get auto-merged - can you approve if it looks okay?
